gdnative-core/src/nativescript/init/property/hint.rs

```diff
@@ -9,6 +9,17 @@ use crate::sys;
 
 use super::ExportInfo;
 
+/// Need the Hint trait so that nested Array hints work.
+pub trait Hint: fmt::Debug {
+    fn export_info(&self) -> ExportInfo;
+}
+
+impl Hint for () {
+    fn export_info(&self) -> ExportInfo {
+        panic!("Unit hint should never be called, only used as a hintless placeholder.");
+    }
+}
+
```

Comment:
This can be avoided by using `Export` as the trait bound in `ArrayHint::with_subhint` instead.